crates/polars-time/src/chunkedarray/rolling_window/rolling_kernels/no_nulls.rs

```diff
@@ -50,6 +53,75 @@ where
     Ok(Box::new(out))
 }
 
+// Use an aggregation window that maintains the state
+pub(crate) fn rolling_apply_agg_window<'a, Agg, T, O>(
+    values: &'a [T],
+    offsets: O,
+    min_periods: usize,
+    params: DynArgs,
+    sorting_indices: Option<&[IdxSize]>,
+) -> PolarsResult<ArrayRef>
+where
+    // items (offset, len) -> so offsets are offset, offset + len
+    Agg: RollingAggWindowNoNulls<'a, T>,
+    O: Iterator<Item = PolarsResult<(IdxSize, IdxSize)>> + TrustedLen,
+    T: Debug + IsFloat + NativeType,
+{
+    if values.is_empty() {
+        let out: Vec<T> = vec![];
+        return Ok(Box::new(PrimitiveArray::new(
+            T::PRIMITIVE.into(),
+            out.into(),
+            None,
+        )));
+    }
+    let sorting_indices = sorting_indices.expect("`sorting_indices` should have been set");
+    // start with a dummy index, will be overwritten on first iteration.
+    let mut agg_window = Agg::new(values, 0, 0, params);
+
+    let mut out = zeroed_vec(values.len());
```

Comment:
Hi @orlp - I've tried using `zeroed_vec` here, and creating a validity mask separately. Any feedback on how I've done this would be appreciated if possible, thanks 🙏